### PR TITLE
Update DG content per required

### PR DESCRIPTION
**Pull request by Diwu-Yi**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -16,6 +15,12 @@ optimised for the Command Line Interface (CLI), so that users can update and ret
 ### Purpose of Document
 This document specifies the architecture and software design for the application, tCheck.
 
+### Audience
+
+The Developer Guide is designed for those who are interested in understanding the architecture and other aspects of software design
+of tCheck. In particular, this guide has been written with the current and future tCheck developers in mind because it details
+the knowledge necessary to know to be able to modify the codebase and customize tCheck for specific operational needs or extend current functionalities.
```
Comment (ureshiiYing) [inline docs/DeveloperGuide.md]: Incorrect grammar. "... knowledge necessary to (be able to) modify ... "
The words in brackets are optional, you may add it or not. If you do, remove the brackets.

"customise" for british spelling.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -876,6 +896,14 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
 * **Private contact detail**: A contact detail that is not meant to be shared with others
```
Comment (ureshiiYing) [inline docs/DeveloperGuide.md]: Please remove this.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Diwu-Yi) [reply docs/DeveloperGuide.md]: Noted, will change

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Diwu-Yi) [reply docs/DeveloperGuide.md]: Noted, will remove

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -803,25 +812,37 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  Steps 2b1-2b2 are repeated until a valid amount is entered.
 
   Use case ends.
 
 * 2c. tCheck detects missing field(s) in the command entered.
 
-  * 2c1. tCheck displays an error message informing the user that there are missing field(s) which caused the command to fail and shows an example of a correct command.
+  * 2c1. tCheck requests the user to re-enter the command with all necessary fields.
+  
+  * 2c2. User enters new a command with necessary fields.
```
Comment (WM71811) [inline docs/DeveloperGuide.md]: a new command?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -803,25 +812,37 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-        Use case ends.
+  Use case ends.
 
 **Extensions**
 
 * 2a. tCheck is unable to find the entered ingredient name.
 
-  * 2a1. tCheck displays an error message informing the user that the ingredient name entered is not found in the pre-defined ingredient book.
+  * 2a1. tCheck requests the user to re-enter the command with correct ingredient name.
```
Comment (WM71811) [inline docs/DeveloperGuide.md]: the/a correct ingredient name?

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Diwu-Yi) [reply docs/DeveloperGuide.md]: Noted, thanks

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (Diwu-Yi) [reply docs/DeveloperGuide.md]: Noted, thanks

**Review by yanbingtao — APPROVED** [review]
LGTM

**Review by ureshiiYing — APPROVED** [review]
LGTM

**Pull request merged**